soap/client.go

```diff
@@ -96,6 +91,31 @@ func (c *Client) RoundTrip(in, out Message) error {
 	return xml.NewDecoder(resp.Body).Decode(out)
 }
 
+// RoundTrip implements the RoundTripper interface.
+func (c *Client) RoundTrip(in, out Message) error {
+	headerFunc := func(r *http.Request) {
+		ct := c.ContentType
+		if ct == "" {
+			ct = "text/xml"
+		}
+		r.Header.Set("Content-Type", ct)
+		// TODO - what should happen here? there is a test case that does send in a nil & it was triggering a panic,
```

Comment:
I think it has been (at least it does locally on revision 40fd89a0a40377adf9b1baaf24a37fa7fc67be45) - regardless left the check in & removed the comment